pkg/envd/docker.go

```diff
@@ -193,6 +194,31 @@ func (e dockerEngine) ListEnvPortBinding(ctx context.Context, env string) ([]typ
 		return nil, errors.Wrap(err, "failed to get container")
 	}
 	ports := types.NewPortBindingFromContainerJSON(ctr)
+	images, err := e.ListImage(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to list image")
+	}
+
+	var portMap = make(map[string]string)
+	for _, image := range images {
+		if strings.HasPrefix(image.Name, fmt.Sprintf("%s:", env)) {
+			var environmentPorts []servertypes.EnvironmentPort
+			err := json.Unmarshal([]byte(image.Labels[types.ImageLabelPorts]), &environmentPorts)
+			if err != nil {
+				return nil, errors.Wrap(err, "Error parsing environment ports")
+			}
+
+			for _, environmentPort := range environmentPorts {
+				portMap[fmt.Sprint(environmentPort.Port)] = environmentPort.Name
+			}
+		}
+	}
+
+	for index, label := range ports {
+		label.Name = portMap[label.Port]
+		ports[index] = label
+	}
```

Comment:
Can you get the same information from `ctr, err := e.ContainerInspect(ctx, env)`? If so, I think you should add some logic to `NewPortBindingFromContainerJSON`.

BTW, if you modify the `dockerEngine.ListEnvPortBinding`, you should also modify `envdServerEngine.ListEnvPortBinding`.

Reply:
I remember looking into `ContainerInspect` and not finding any label/name info, I will look again. I will also modify `envdServerEngine.ListEnv...`

Reply:
I found label info in `ctr` 😓 and moved the logic into `NewPortBindingFromContainerJSON`